src/application/services/useNoteSettings.ts

```diff
@@ -65,14 +65,21 @@ export default function (): UseNoteSettingsComposableState {
   };
 
   /**
-   * Update note settings
+   * Update field isPublic in note settings
    *
    * @param id - Note id
-   * @param data - Note settings data with new values
+   * @param newIsPublicValue - new isPublic
    */
-  const update = async (id: NoteId, data: Partial<NoteSettings>): Promise<void> => {
-    noteSettings.value = await noteSettingsService.patchNoteSettingsByNoteId(id, data);
-  };
+  async function updateIsPublic(id: NoteId, newIsPublicValue: boolean): Promise<void> {
+    const { isPublic } = await noteSettingsService.patchNoteSettingsByNoteId(id, { isPublic: newIsPublicValue });
+
+    /**
+     * Check in case noteSettings has not yet been created and is set to null
```

Comment:
> noteSettings has not yet been created

it's impossible. Note Settings are being created along with note.

Reply:
Then maybe just Check in case noteSettings is set to null?

Reply:
I'm suggesting to leave the code as is, but improve docs describing a real-world case